src/Microsoft.Health.Extensions.Xunit/FixtureArgumentSetsXunitTestFrameworkDiscoverer.cs

```diff
@@ -0,0 +1,144 @@
+
+            foreach (var method in testClass.Class.GetMethods(true))
+            {
+                IAttributeInfo fixtureParameterAttributeInfo = method.GetCustomAttributes(typeof(FixtureArgumentSetsAttribute)).SingleOrDefault();
+
+                SingleFlagEnum[][] closedSets = classLevelClosedParameterSets;
+
+                if (fixtureParameterAttributeInfo != null)
+                {
+                    // get the method-level parameter sets in the form (Arg1.OptionA, Arg1.OptionB), (Arg2.OptionA, Arg2.OptionB)
+                    SingleFlagEnum[][] methodLevelOpenParameterSets = ExpandEnumFlagsFromAttributeData(fixtureParameterAttributeInfo);
+
+                    bool hasOverride = false;
+                    for (int i = 0; i < methodLevelOpenParameterSets.Length; i++)
+                    {
+                        if (methodLevelOpenParameterSets[i]?.Length > 0)
+                        {
+                            hasOverride = true;
+                        }
+                        else
+                        {
+                            // means take the class-level set
+                            methodLevelOpenParameterSets[i] = classLevelOpenParameterSets[i];
+                        }
+                    }
+
+                    if (hasOverride)
+                    {
+                        // convert to the form (Arg1.OptionA, Arg2.OptionA), (Arg1.OptionA, Arg2.OptionB), (Arg1.OptionB, Arg2.OptionA), (Arg1.OptionB, Arg2.OptionB)
+                        closedSets = CartesianProduct(methodLevelOpenParameterSets).Select(e => e.ToArray()).ToArray();
+                    }
+                }
+
+                foreach (SingleFlagEnum[] closedVariant in closedSets)
+                {
+                    var closedVariantTestClass = new TestClassWithFixtureArguments(testClass.TestCollection, testClass.Class, closedVariant);
+                    var closedVariantTestMethod = new TestMethod(closedVariantTestClass, method);
+
+                    if (!FindTestsForMethod(closedVariantTestMethod, includeSourceInformation, messageBus, discoveryOptions))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        private static SingleFlagEnum[][] ExpandEnumFlagsFromAttributeData(IAttributeInfo attributeInfo)
+        {
+            bool IsPowerOfTwo(long x)
+            {
+                return (x != 0) && ((x & (x - 1)) == 0);
+            }
+
+            IEnumerable<SingleFlagEnum> GetSingleValuedFlags(Enum e)
+            {
+                if (e is null)
+                {
+                    yield break;
+                }
+
+                var enumAsLong = Convert.ToInt64(e);
+
+                foreach (Enum value in Enum.GetValues(e.GetType()))
+                {
+                    var flagAsLong = Convert.ToInt64(value);
+                    if (IsPowerOfTwo(flagAsLong))
```

Comment:
Is there ever a case where the `value` won't be a power of 2 here?

Reply:
Yes, it you have an enum like this
``` csharp
enum 
{
  A = 1,
  B = 2,
  All = A | B
}
```